ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -619,6 +592,14 @@ public void validate() {
     }
 
     private void addSelectItem(final Expression exp, final ColumnName columnName) {
+      if (persistent) {
+        if (SchemaUtil.ROWTIME_NAME.equals(columnName)
+            || SchemaUtil.ROWKEY_NAME.equals(columnName)) {
+          throw new KsqlException("Reserved column name in select: " + columnName + ". "
+              + "Please remove or alias the column.");
```

Comment:
I think we should be even stricter than just requiring an alias - it should be a totally different name (i.e. `A_ROWKEY`). Otherwise I'm a little confused about why this works:
```sql
CREATE STREAM B AS SELECT A.FOO, A.ROWKEY FROM A PARTITION BY A.BAR;
```
I should then be able to `SELECT FOO, ROWKEY FROM B` and get the old values of `FOO` and `BAR` - not the old `ROWKEY`

Reply:
Erm... that's what it's requiring....

What its not allowing is for their to be a column selection/projection with the _reserved_ column name of `ROWKEY` or `ROWTIME`.   While this makes sense for a transient query, (which only return the value columns at the moment), such columns should not be allowed in persistent queries until we fix the query semantics.

However, if you alias the column it no longer has a reserved column name, so is allowed, e.g.

```json
{
      "name": "non-join leaves aliased ROWKEY and ROWTIME in output's value schema",
      "statements": [
        "CREATE STREAM INPUT (F0 INT) WITH (kafka_topic='input', value_format='JSON');",
        "CREATE STREAM OUTPUT AS SELECT F0, ROWTIME AS TIME, ROWKEY AS KEY FROM INPUT;"
      ],
      "inputs": [{"topic": "input", "key": "k", "value": {"F0": 4}, "timestamp": 1}],
      "outputs": [{"topic": "OUTPUT", "key": "k", "value": {"F0": 4, "TIME": 1, "KEY": "k"}, "timestamp": 1}]
    }
```

For your example above:

```sql
CREATE STREAM B AS SELECT A.FOO, A.ROWKEY FROM A PARTITION BY A.BAR;
```

if its allowed, (I've not checked), its because `A.ROWKEY` results in a column called 
`A_ROWKEY`, which is not a reserved column name.

How that column name is made is up for debate, but, honestly, outside the scope of this PR.